dimod/binary_quadratic_model.py

```diff
@@ -59,15 +59,58 @@
 
 from six import itervalues, iteritems
 
-from dimod.decorators import vartype_argument
+from dimod.decorators import vartype_argument, lockable_method
 from dimod.serialization.utils import array2bytes, bytes2array
 from dimod.sampleset import as_samples
 from dimod.utilities import resolve_label_conflict
 from dimod.views.bqm import LinearView, QuadraticView, AdjacencyView
 from dimod.views.samples import SampleView
 from dimod.vartypes import Vartype
 
-__all__ = 'BinaryQuadraticModel', 'BQM'
+__all__ = ['BinaryQuadraticModel', 'BQM']
+
+
+class LockableDict(dict):
+    """A dict that can turn writeablity on and off"""
+
+    # methods like update, clear etc are not wrappers for __setitem__,
+    # __delitem__ so they need to be overwritten
+
+    @property
+    def is_writeable(self):
+        return getattr(self, '_writeable', True)
+
+    @is_writeable.setter
+    def is_writeable(self, b):
+        self._writeable = bool(b)
```

Comment:
Why complicate things with wrapping `_writeable`, when you can just have `writeable` class/member variable (or even `is_writeable`)?